### ISSUE TITLE
Callback does not work as documented

### ISSUE THREAD

**Issue by DDN-Shep** [gulpjs/gulp#1914]
**What were you expecting to happen?**
I expected the watch cb() to be called once a .js file was added to the ./dist folder.

As documented here;
[gulp watch cb(event)](https://github.com/gulpjs/gulp/blob/master/docs/API.md#cbevent)

**What actually happened?**
File added events do not trigger a callback, only file changed events work.

```js
gulp.watch('./dist/*.js', (e) => {
  console.log('Watch event type', e.type); // Does not work for 'Added' events, only 'Changed' events.
  // do stuff...
});
```

**What version of gulp are you using?**
CLI version 1.2.2
Local version 3.9.1

**What versions of npm and node are you using?**
NPM version 3.10.10
Node version 7.2.1

**Comment by DDN-Shep** [comment]
Further to this, I have found that the dependency on Gaze may actually be the problem here...

**Comment by ericmdantas** [comment]
See this: https://github.com/gulpjs/gulp/issues/651

**Comment by ericmdantas** [comment]
Also, [gulp-watch](https://github.com/floatdrop/gulp-watch) might help you accomplish what you want.

**Comment by DDN-Shep** [comment]
Thanks, I did have a brief look at that module, but in the end, have gone with a workaround for now.

FYI, I tried using Gaze directly (with different GLOB patterns just in case), but the 'added' event type is never triggered.

My current workaround is to ensure there is a file there, before running the watch task;

```
// Workaround for Gaze bug ('added' events are not triggered, only 'changed' events)
fs.closeSync(fs.openSync('./dist/myfile.min.js', 'w'));
```

Not ideal, but it looks like the issue is with Gaze, and not gulp.watch.
